src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/Methodologies/MethodologyServiceTests.cs

```diff
@@ -980,9 +1081,261 @@ await using (var context = InMemoryApplicationDbContext(contentDbContextId))
+                    cacheService: cacheService.Object,
+                    methodologyContentService: contentService.Object,
+                    methodologyImageService: imageService.Object,
+                    methodologyRepository: methodologyRepository.Object,
+                    publishingService: publishingService.Object);
+
+                var result = await service.UpdateMethodology(methodology.Id, request);
+                result.AssertNotFound();
+            }
+
+            VerifyAllMocks(cacheService, contentService, imageService, methodologyRepository, publishingService);
+        }
+
+        [Fact]
+        public async Task UpdateMethodology_ApprovingUsingWithReleaseStrategy_ReleaseAlreadyPublished()
+        {
+            var publication = new Publication();
+
+            // Create a release that is already published which the methodology cannot be made dependant on
+            var scheduledWithRelease = new Release
+            {
+                Id = Guid.NewGuid(),
+                Publication = publication,
+                TimePeriodCoverage = CalendarYear,
+                ReleaseName = "2021",
+                Published = DateTime.UtcNow
+            };
+
+            var methodology = new Methodology
+            {
+                PublishingStrategy = Immediately,
+                Status = Draft,
+                MethodologyParent = new MethodologyParent
+                {
+                    Publications = AsList(new PublicationMethodology
+                    {
+                        Owner = true,
+                        Publication = publication
+                    })
+                }
+            };
+
+            var request = new MethodologyUpdateRequest
+            {
+                LatestInternalReleaseNote = "Test approval",
+                PublishingStrategy = WithRelease,
+                WithReleaseId = scheduledWithRelease.Id,
+                Status = Approved,
+                Title = "Title"
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var context = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await context.Methodologies.AddAsync(methodology);
+                await context.Releases.AddAsync(scheduledWithRelease);
+                await context.SaveChangesAsync();
+            }
+
+            var cacheService = new Mock<ICacheService>(Strict);
+            var contentService = new Mock<IMethodologyContentService>(Strict);
+            var imageService = new Mock<IMethodologyImageService>(Strict);
+            var methodologyRepository = new Mock<IMethodologyRepository>(Strict);
+            var publishingService = new Mock<IPublishingService>(Strict);
+
+            await using (var context = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var service = SetupMethodologyService(contentDbContext: context,
```

Comment:
I'm tempted to say that rather than having to create all of these Mocks each time as Strict and then pass them into `SetupXxxxxService`, instead the default values that we construct in `SetupXxxxxService` should be strict.  As we're not setting up any calls on them in these cases, there's no need to have them available to pass into VerifyAllMocks()

Reply:
That's a good idea, and it will remove quite a lot of unecessary setup. I've made this change now.